crates/router/src/core/payments/routing.rs

```diff
@@ -868,7 +875,139 @@ pub async fn perform_session_flow_routing(
     transaction_type: &api_enums::TransactionType,
 ) -> RoutingResult<FxHashMap<api_enums::PaymentMethodType, Vec<routing_types::SessionRoutingChoice>>>
 {
-    todo!()
+    let mut pm_type_map: FxHashMap<api_enums::PaymentMethodType, FxHashMap<String, api::GetToken>> =
+        FxHashMap::default();
+
+    #[cfg(feature = "v2")]
+    let profile_id = session_input.payment_intent.profile_id.clone();
+
+    let business_profile = session_input
```

Comment:
We should not read business_profile deep inside here, read in parent function which defined in payments and get as argument.